File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Palau.m
//
//  UIImage+Palau.m
//  CountriesFlags
//

#import "Countries/UIImage+Palau.h"

@implementation UIImage (Palau)

+ (UIImage *)flagPalauWithSize:(CGSize)size {
    UIColor* colorBlue = [UIColor colorWithRed:0.0f
                                         green:133.0f/255.0f
                                          blue:202.0f/255.0f
                                         alpha:1.0f];
    UIColor* colorYellow = [UIColor colorWithRed:1.0f
                                           green:209.0f/255.0f
                                            blue:0.0f
                                           alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorYellow setFill];
    [[UIBezierPath bezierPathWithArcCenter:CGPointMake(size.width * 7.0f / 16.0f, size.height / 2.0f)
                                    radius:size.height * 3.0f / 10.0f
                                startAngle:0.0f
                                  endAngle:M_PI * 2.0f
                                 clockwise:YES] fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image != nil ? image : [UIImage new];
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Palau.swift
//
//  UIImage+Palau.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Palau "wikipedia"
     
     Generates the flag of Palau.
     For more information see [wikipedia].
     
     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Palau.
     */
    internal static func flagPalau(with size: CGSize) -> UIImage? {
        let colorBlue = UIColor(red: 0.0,
                                green: 133.0 / 255.0,
                                blue: 202.0 / 255.0,
                                alpha: 1.0)
        let colorYellow = UIColor(red: 1.0,
                                  green: 209.0 / 255.0,
                                  blue: 0.0,
                                  alpha: 1.0)

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorBlue.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        let radius = size.height * 3.0 / 5.0
        let path = UIBezierPath(ovalIn: CGRect(x: size.width * 7.0 / 16.0 - radius / 2.0,
                                               y: size.height / 2.0 - radius / 2.0,
                                               width: radius,
                                               height: radius))
        colorYellow.setFill()
        path.fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
